### DISCUSSION TITLE
GUT for ArcGIS Pro

### DISCUSSION THREAD

**Discussion by CDawson20** [Riverscapes/pyGUT#10] [Q&A]
I have been using the GUT for several years through ArcMap. When I try to run the same files on my institutional machine (ArcGIS Pro), I receive a "SyntaxError: invalid syntax".  Is the GUT compatible with Pro?
![error](https://github.com/Riverscapes/pyGUT/assets/170964257/665412af-5f58-4f84-9ff0-fe155f0feb9c)

**Comment by cmac95** [comment]
@CDawson20 - Geomorphic Unit Tool (GUT) has been written in Python 2.7, which as you mention is (was) associated with ArcGIS Desktop (ArcMap etc.). This software is rapidly being retired, and is now in mature support. 
https://support.esri.com/en-us/products/arcmap/life-cycle

The error you are getting will be because some of the syntax (likely <>) has been updated as ArcGIS Pro uses Python 3. 

I have raised this issue on this repo, however have not received any feedback from the developers yet. I am also toying with the idea of just re-writing the script myself, but that takes time and would need some funding in my current employment. 

I have also mentioned this to some folks at ESRI which I will follow-up on shortly.

Craig.

**Reply by CDawson20** [reply]
Hi Craig,
I have re-written the config and teirFunctions scripts for Py3 (attached).

[config1 - Copy.txt](https://github.com/user-attachments/files/20119120/config1.-.Copy.txt)
[tierFunctions - Copy.txt](https://github.com/user-attachments/files/20119122/tierFunctions.-.Copy.txt)

Let me know if you have trouble downloading, I can send you my email.

Thank you for your feedback!
Corey
